### DISCUSSION TITLE
Dimension information of the outputs

### DISCUSSION THREAD

**Discussion by cassiemavs** [NREL/ReEDS-2.0#155] [Q&A: Analysis] [answered]
Is there any documation about the dimension names for the output? For example, I'm looking at the variable cap_new_bin_out in the output gdx file (or cap_new_bin_out.csv). I can identify the first column (e.g., battery_2) is the fuel technology, thrid column (e.g., p1) is the BA region, four column is the year. But I don't know what the second column (e.g., new1, new33) and fifth column (e.g., bin1, bin 18) are. Any advice on this?

**Comment by wesleyjcole** [comment, marked as answer]
The outputs lose their names when they are written to csv files, but you can see them in GAMS.  The e_report.gms file includes all of the outputs that are written by the model.  It will show their indices, units, and how they are calculated.  For example, if you search e_report.gms for cap_new_bin_out, you'll see this:

`cap_new_bin_out(i,v,r,t,rscbin)      "--MW-- capacity of built techs"`

You'll also see the how cap_new_bin_out is calculated:

```
cap_new_bin_out(i,v,r,t,rscbin)$[rsc_i(i)$valinv(i,v,r,t)] = INV_RSC.l(i,v,r,rscbin,t) / ilr(i) ;
cap_new_bin_out(i,v,r,t,"bin1")$[(not rsc_i(i))$valinv(i,v,r,t)] = INV.l(i,v,r,t) / ilr(i) ;
```

If you need to know what specific indices are in GAMS, you can search for those to identify them upstream.  For example, a search for "v" will lead you to this in b_inputs.gms:

```
* Each generation technology is broken out by class:
* 1. initial capacity: init-1, init-2, ..., init-n
* 2. prescribed capacity: prescribed
* 3. new capacity: new
* This allows us to distinguish between existing, prescribed, and model-chosen builds
* The number of classes is set by numhintage for initial capacity and numclass for new capacity
v "technology class"
    /
      init-1*init-%numhintage%,
      new1*new%numclass%
    /,
```

**Reply by cassiemavs** [reply]
Thank you for the quick response! This information is really helpful.
